# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1287814976 14400
#      Sat Oct 23 02:22:56 2010 -0400
# Node ID 69b2f237060e9209e6334add5a177a85752a3de2
# Parent  3735abe5ebbe7f1a2ebd610e905243b676291e89
file-io.cc (Ffopen): argument parsing tweak

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2010-10-23  John W. Eaton  <jwe@octave.org>
+
+	* file-io.cc (Ffopen): Only handle fopen ("all") case if nargout
+	is 0 or 1.
+
 2010-10-22  John W. Eaton  <jwe@octave.org>
 
 	* octave.cc (usage): Put whitespace before and after usage message.
 	(octave_main): If getopt_long returns '?', it means an
 	unrecognized option was encountered.
 	Panic if default case in option switch statement is reached.
  	(verbose_usage, short_opts, usage_string): Remove '?' as an
 	alias for 'h' in the list of possible options.
diff --git a/src/file-io.cc b/src/file-io.cc
--- a/src/file-io.cc
+++ b/src/file-io.cc
@@ -549,17 +549,17 @@ do_stream_open (const octave_value& tc_n
         ::error ("%s: file mode must be a string", fcn);
     }
   else
     ::error ("%s: file name must be a string", fcn);
 
   return retval;
 }
 
-DEFUN (fopen, args, ,
+DEFUN (fopen, args, nargout,
   "-*- texinfo -*-\n\
 @deftypefn  {Built-in Function} {[@var{fid}, @var{msg}] =} fopen (@var{name}, @var{mode}, @var{arch})\n\
 @deftypefnx {Built-in Function} {@var{fid_list} =} fopen (\"all\")\n\
 @deftypefnx {Built-in Function} {[@var{file}, @var{mode}, @var{arch}] =} fopen (@var{fid})\n\
 The first form of the @code{fopen} function opens the named file with\n\
 the specified mode (read-write, read-only, etc.) and architecture\n\
 interpretation (IEEE big endian, IEEE little endian, etc.), and returns\n\
 an integer value that may be used to refer to the file later.  If an\n\
@@ -659,17 +659,17 @@ however, conversions are currently only 
   octave_value_list retval;
 
   retval(0) = -1.0;
 
   int nargin = args.length ();
 
   if (nargin == 1)
     {
-      if (args(0).is_string ())
+      if (nargout < 2 && args(0).is_string ())
         {
           // If there is only one argument and it is a string but it
           // is not the string "all", we assume it is a file to open
           // with MODE = "r".  To open a file called "all", you have
           // to supply more than one argument.
 
           if (args(0).string_value () == "all")
             return octave_stream_list::open_file_numbers ();
